clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -494,6 +495,16 @@ private long waitOnMetadata(String topic, long maxWaitMs) throws InterruptedExce
         return time.milliseconds() - begin;
     }
 
+    private long getTimestamp(String topic, Long timestamp) {
+        // If log append times is used for the topic, we overwrite the timestamp to avoid server side re-compression.
+        if (metadata.isUsingLogAppendTime(topic))
```

Comment:
Related to my other comment about learning about timestamp type for topic. So, the first set of produce messages will not have timestamp == INHERITED_TIMESTAMP if timestamp type == LogAppendTime, right? If setting timestamp to INHERITED_TIMESTAMP  is required for compressed messages to work, does it mean we have a bug?